client/state/happychat/middleware.js

```diff
@@ -67,6 +132,14 @@ export default function( connection = null ) {
 
 	return store => next => action => {
 		switch ( action.type ) {
+			case HAPPYCHAT_CONNECTING:
+				if ( getHappychatConnectionStatus( store.getState() ) === 'connected' ) {
+					// If chat is already connected, do nothing and stop the action from proceeding to reducers
+					return;
```

Comment:
At the first  glance, I thought this would be `return next( action );` or `break;` to continue the action processing chain. However, from the test suite it seems to be intended. Could you help me understand the reason behind this?